pkgs/applications/science/logic/rocq/default.nix

```diff
@@ -0,0 +1,152 @@
+# - The csdp program used for the Micromega tactic is statically referenced.
+#   However, rocq can build without csdp by setting it to null.
+#   In this case some Micromega tactics will search the user's path for the csdp program and will fail if it is not found.
+# - The exact version can be specified through the `version` argument to
+#   the derivation; it defaults to the latest stable version.
+
+{ lib, stdenv, fetchzip, fetchurl, writeText, pkg-config, gnumake42
+, customOCamlPackages ? null
+, ocamlPackages_4_14
```

Comment:
Naïve question: would it be better to use the current version of OCaml, i.e., 5.2 (and maybe soon 5.3)? or is OCaml 4.14 the correct thing to use to compile Rocq as of today?

This question is both for the PR author and for any careful reader such as Théo.

Reply:
OCaml 5 has the following disadvantages:
* not native_compute
* small performance penalty

My understanding is that nixpkgs doesn't really support native_compute anyway so maybe we could switch to OCaml 5. No strong opinion here.

Reply:
> small performance penalty

I'm unsure that this is still true with OCaml 5.2.



> My understanding is that nixpkgs doesn't really support native_compute anyway so maybe we could switch to OCaml 5. No strong opinion here.

Indeed! Native compute users could still override the version of OCaml to use OCaml 4.14 instead.

Reply:
> > small performance penalty
> 
> I'm unsure that this is still true with OCaml 5.2.

My understanding is that it will "always" remain true for fundamental reasons. But I may be wrong.